recipes-test/demo-config/primary-config.bb

```diff
@@ -11,6 +11,7 @@ PRIMARY_SECONDARIES ?= "${SECONDARY_IP}:${SECONDARY_PORT}"
 SRC_URI = "\
     file://30-secondary-config.toml \
     file://ip_secondary_config.json \
+    ${@('file://' + d.getVar('SOTA_SECONDARY_CONFIG')) if d.getVar('SOTA_SECONDARY_CONFIG') else ''} \
```

Comment:
I am not sure if I get what is this for?

Reply:
bitbake needs the file in `SRC_URI` to track a dependency.

Without this change, if you build an image then modify the file and rebuild, the new file version won't be taken into account and bitbake will just give you the same image from cache

With this change, the image is rebuilt as expected.

Reply:
ok, thanks for the explanation.

Reply:
ok, thanks for the explanation!